doc/architecture/decisions/0031-replace-publicactivity-with-our-own-hand-rolled-way-of-recording-changes.md

```diff
@@ -0,0 +1,25 @@
+# 31. Replace PublicActivity with our own hand-rolled way of recording changes
+
+Date: 2021-06-30
+
+## Status
+
+Accepted
+
+## Context
+
+We adopted the `PublicActivity` gem (as documented in [0019](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/731b91f5773c4306f919433af81976d29c5feef1/doc/architecture/decisions/0019-use-public-activity-gem.md)), but the information we record has never been used, or displayed to the user.
+
+Now we're looking at surfacing information about changes to end users it's becoming increasingly clear that is `PublicActivity` is not fit for our needs, and would be easier to build our own approach to logging what changed were made by what user.
+
+## Decision
+
+We have decided to introduce a `HistoricalEvent` model to record changes to models, starting with `Activity`s.
+
+Once we have applied `HistoricalEvents` to all models, we can remove all the calls to `PublicActivity` and remove the gem entirely.
```

Comment:
Maybe something like "applied `HistoricalEvents` to the models we need to track, we can remove..." to make it clear that we don't wish to track changes to *all* models / entities?

Reply:
Good call 👍